sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/visitors/outer/monotonicity/Monotonicity.java

```diff
@@ -535,6 +537,37 @@ public void postorder(DBSPWindowOperator node) {
         this.set(node, output);
     }
 
+    @Override
+    public void postorder(DBSPLagOperator node) {
+        MonotoneExpression input = this.getMonotoneExpression(node.input());
+        if (input == null)
+            return;
+        IMaybeMonotoneType projection = Monotonicity.getBodyType(input);
+        // The lag operator function has 2 inputs, although the operator has
+        // a single input.  The first parameter is actually fed from the
+        // operator's input.  The second parameter is fed from the lagged version of
```

Comment:
which isn't the same as delayed input in DBSP (both inputs are extracted from the latest integral of the stream). Just clarifying...